### DISCUSSION TITLE
Prometheus metrics

### DISCUSSION THREAD

**Discussion by git-josip** [apache/camel-karavan#822] [Q&A] [answered]
I am using a Karavan k8s spring boot. I am wondering did anyone tried to set up and expose Prometheus metrics so they can be collected.
I know spring boot can be configured to expose Prometheus metrics in the actuator and deployed k8s service or pod configured via Prometheus ServiceMonitor so metrics are collected automatically.

**Comment by davsclaus** [comment, marked as answer]
Yeah you need to do all of this the standard spring boot way - add some prometheus -starter as dependency and maybe some configuration into application.properties.

And add camel-micrometer-starter then it auto collects metrics and via a micrometer-promethus (spring boot standard) then it can be exposed over http in prometheus protcol
https://camel.apache.org/components/3.20.x/micrometer-component.html#_spring_boot_auto_configuration

In karavan, you can just add camel-micrometer as dependency

**Comment by git-josip** [comment]
@davsclaus thank you for your reply. I was able to enable prometheus metrics actuator endpoint. 
First I added in proeprties: 
`camel.jbang.dependencies=camel:microprofile-health,mvn:io.micrometer:micrometer-registry-prometheus:1.11.1` 
and after that enabled actuator prometheus endpoint: 
`management.endpoints.web.exposure.include=health,info,prometheus,configprops,env` 

I have next question, when my pod is running with metrics exposed I need to dynamically tell prometheus to scrape it and I would like to do that using Prometheus operator.  I would like to add custom label that wil be there for all deployed pods and then I will use that label to tell PrometheusOperator to scrape. 

CUrrent pod has these labels: 
```
Labels:        
                  app=hzv
                  app.kubernetes.io/runtime=camel
                  group=org.camel.karavan.demo
                  pod-template-hash=898cc4c66
                  provider=jkube
                  version=1
```

**Reply by davsclaus** [reply]
You can use jkube fragments, or set labels via `label.xxx=aaa` in the application properties
